docs/transaction-intent.md

```diff
@@ -0,0 +1,31 @@
+# TransactionIntent
```

Comment:
I change the intent object for `TransactionIntent`. It seems a name more aligned with the rest of the API. What do you think?